### PR TITLE
AWS  vpc support 

### PR DESCRIPTION
**Pull request by eschultink**
### Features
 - support putting Lambdas on VPC (beta)

### Change implications

 - dependencies added/changed? **yes** - use of AWS VPC depends on VPC 
 - something important to note in future release notes?
   - NOTE in `CHANGELOG.md` anything that will show up in `terraform plan`/`apply` that isn't
     obviously a no-op?
   - breaking changes? if in module/example that is NOT marked `alpha`, requires major version
     change

## TODO
 - [ ] ~~move policy down to each connector~~ - *save for beta*
 - [x] CMEK for apigateway log group

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206296312465377

### CODE REVIEW

**Comment by eschultink** [comment]
problem in lambda's `init` phase, although seems unrelated vpc/gateway changes ... need to debug

File: infra/examples-dev/aws-all/variables.tf [hunk 1]
```diff
@@ -186,6 +186,11 @@ variable "custom_api_connector_rules" {
   default     = {}
 }
 
+variable "use_api_gateway" {
```
Comment (aperez-worklytics) [inline infra/examples-dev/aws-all/variables.tf:189]: Rename to `use_api_gateway_v2` to make it clear that is for v2 ?

File: infra/modules/aws-psoxy-rest/main.tf [hunk 2]
```diff
@@ -63,9 +70,56 @@ resource "aws_lambda_function_url" "lambda_url" {
...
+  payload_format_version = "2.0"
+
+  # q: do we need to specify this?? what's the default?
+  # docs saw it can be 30000 max, but ideally we'd get ~55 or longer
+  # timeout_milliseconds   = 30000
+}
+
+resource "aws_apigatewayv2_route" "methods" {
+  # TODO: allow configuration of methods
+  for_each = toset(local.use_api_gateway ? ["GET", "HEAD"] : [])
```
Comment (aperez-worklytics) [inline infra/modules/aws-psoxy-rest/main.tf:91]: We will need `POST` to for GH and Dropbox

File: infra/modules/aws/main.tf [hunk 3]
```diff
@@ -144,6 +148,67 @@ module "psoxy_package" {
...
+
+resource "aws_iam_policy" "invoke_api" {
+  count = var.use_api_gateway ? 1 : 0
+
+  name_prefix = "${var.deployment_id}InvokeAPI"
+
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
```
Comment (aperez-worklytics) [inline infra/modules/aws/main.tf:180]: Yep, missing post for policies too

**Review by aperez-worklytics — CHANGES_REQUESTED** [review]

File: infra/examples-dev/aws-all/variables.tf (hunk 1 above)
Comment (eschultink) [reply infra/examples-dev/aws-all/variables.tf:189]: 🤔  feels hacky to me to expose version numbers, but OK

Comment (aperez-worklytics) [reply infra/examples-dev/aws-all/variables.tf]: Agree, it was a suggestion, but on the other hand we had `use_api_gateway` and then in the description we enforce that is *enable api gateway v2* (as opposite to use v1), so yeah, agree that is weird to expose version numbers...

File: infra/modules/aws/main.tf [hunk 4]
```diff
@@ -197,3 +267,14 @@ output "pseudonym_salt" {
   value       = random_password.pseudonym_salt.result
   sensitive   = true
 }
+
+output "api_gateway_v2" {
+  # NOTE: filled based on `var.use_api_gateway`, which is sufficient for Terraform to understand
```
Comment (aperez-worklytics) [inline infra/modules/aws/main.tf:272]:
```suggestion
  # NOTE: filled based on `var.use_api_gateway_v2`, which is sufficient for Terraform to understand
```

**Review by aperez-worklytics — APPROVED** [review]

**Pull request merged**